frontend/src/Frontend/Examples/WebSocketChat/Main.hs

```diff
@@ -61,10 +66,22 @@ app = do
           el "div" $ text err
           return never
         Right encoder -> do
-          let wsPath = T.intercalate "/" $ fst $ encode encoder $ InL BackendRoute_WebSocketChat :/ ()
+          let wsPath = fst $ encode encoder $ InL BackendRoute_WebSocketChat :/ ()
               sendEv = fmap ((:[]) . toStrict . Aeson.encode) msgSendEv
-          ws <- webSocket ("ws://localhost:8000/" <> wsPath) $ def & webSocketConfig_send .~ sendEv
-          return (_webSocket_recv ws)
+          r <- liftIO $ Cfg.get "config/common/route"
+          let mUri = do 
+                uri' <- mkURI =<< r 
+                pathPiece <- mapM mkPathPiece wsPath
+                wsScheme <- mkScheme "ws"
```

Comment:
may be we should make use of the existing scheme to determine this?

```
    let Just (URI scheme (Just auth)  _ _ _) = parseURI $ T.unpack $ T.strip r
        wsScheme = case scheme of
          "https:" -> "wss:"
          _ -> "ws:"
```

Reply:
indeed, you should case of `uriScheme uri'` for that